Implement webRTC signalling

shankarsengalani opened the pull request:
Implemented webRTC secure signalling application on SAFE Network.

Commit:
feat/webrtc: implement webrtc signalling

Implemented webRTC secure signalling application on SAFE Network.

maidsafe-highfive commented:
r? @krishnaIndia

(maidsafe_highfive has picked a reviewer for you, use r? to override)

Commit:
fix/credential: remove credential

Removed credentials

Commit:
Chore/readme: update readme.md

Update readme.md for configuration of STUN and TURN server

Commit:
fix/fetch_publicNames: resolve fetch public name issue

Resolve public name issues

bochaco commented on safe_webrtc_example/app/constants.js:
```diff
@@ -0,0 +1,74 @@
+  CONFIG: {
+    SERVER: {
+      iceServers: [
+        { url: 'stun:stun1.l.google.com:19302' }, // URL to STUN Server
```
Perhaps replace the value with a placeholder similar to how it's done with the credential and username.

bochaco commented on safe_webrtc_example/app/constants.js:
```diff
@@ -0,0 +1,74 @@
+      iceServers: [
+        { url: 'stun:stun1.l.google.com:19302' }, // URL to STUN Server
+        {
+          url: 'turn:numb.viagenie.ca',
```
Same comment as above

bochaco commented on safe_webrtc_example/app/constants.js:
```diff
@@ -0,0 +1,74 @@
+  ERR_CODE: {
+    NO_SUCH_ENTRY: -106,
+  },
+  CRYPTO_KEYS: {
```
It seems this can be removed.

bochaco commented on safe_webrtc_example/README.md:
```diff
@@ -0,0 +1,53 @@
+    iceServers: [
+      { url: 'stun:stun1.l.google.com:19302' }, // URL to STUN Server
+      {
+        url: 'turn:numb.viagenie.ca',
```
I'd put placeholders here rather than google stun server url's.

bochaco commented on safe_webrtc_example/app/safe_comm.js:
```diff
@@ -0,0 +1,624 @@
+        resolve(true);
+      } catch (err) {
+        // if (err.code !== CONST.ERR_CODE.NO_SUCH_ENTRY) {
+        if (err.message !== ERROR_MSG.ENTRY_NOT_FOUND) {
```
I'm assuming error codes are not available from the DOM API? If so, I would add a comment (TODO/FIXME) here to state that we should use error codes whenever they become available.

bochaco replied:
I've added some comments for things I'd change, but I thing the most important thing would be that this PR is breaking the CI, apparently just due to a missing `test` script?

Commit:
fix/ux: improve UI/UX

Improved UI/UX features of the application.
Removed server details.
Localised texts.

Commit:
fix/disable_logs: disable logs for production

disable logs for production

Commit:
fix/improve_messages: improve label and message texts

Improved label and message texts to make things consistent throughout the application.

Commit:
fix/remove_credentials: remove credentials

Replace place holder instead of credentials.

Commit:
fix/travis: fix travis issue

Fixed travis issue, failing to find test script on root package.json
